src/dune_rules/mdx.ml

```diff
@@ -365,13 +380,18 @@ let mdx_prog_gen t ~sctx ~dir ~scope ~expander ~mdx_prog =
     in
     S args
   in
-  let prelude_args =
-    Command.Args.S (List.concat_map t.preludes ~f:(Prelude.to_args ~dir))
+  let open Command.Args in
+  let prelude_args = S (List.concat_map t.preludes ~f:(Prelude.to_args ~dir)) in
+  let color_always =
+    if Lazy.force Ansi_color.stderr_supports_color then [ A "--color=always" ]
+    else []
   in
+  let verbose = if !Log.verbose then [ A mdx_verbosity ] else [] in
   (* We call mdx to generate the testing executable source *)
   let action =
     Command.run ~dir:(Path.build dir) mdx_prog ~stdout_to:file
-      [ A "dune-gen"; prelude_args; Resolve.Memo.args directory_args ]
+      ([ A "dune-gen"; prelude_args; Resolve.Memo.args directory_args ]
+      @ color_always @ verbose)
```

Comment:
As the raw result is used to build the executable (it produces OCaml code), I'm not sure the color_always and verbose should be applied here.

Reply:
It does, because mdx logs on stderr and the output is on stdout.